src/algebra/char_p/local_ring.lean

```diff
@@ -0,0 +1,78 @@
+/-
+Copyright (c) 2022 Jon Eugster. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Eugster
+-/
+import algebra.char_p.basic
+import algebra.is_prime_pow
+import data.nat.factorization
+import ring_theory.ideal.local_ring
+
+/-!
+# Characteristics of local rings
+
+## Main result
+
+- `char_p_zero_or_prime_power`: In a commutative local ring the characteristics is either
+  zero or a prime power.
+
+-/
+
+/-- In a local ring the characteristics is either zero or a prime power. -/
+theorem char_p_zero_or_prime_power (R : Type*) [comm_ring R] [local_ring R] (q : ℕ)
+  [char_R_q : char_p R q] : q = 0 ∨ is_prime_pow q :=
+begin
+  /- Assume `q := char(R)` is not zero. -/
+  apply or_iff_not_imp_left.2,
+  intro q_pos,
+  let K := local_ring.residue_field R,
+  haveI RM_char := ring_char.char_p K,
+
+  /- `r := char(R/m)` is either prime or zero. -/
```

Comment:
Can you extract this a sublemma?

Reply:
What exactly would you want as a sublemma?

Reply:
@riccardobrasca based on where this comment is I assume it refers to the comment `r := char(R/m)` is either prime or zero.`. That lemma exists already and is just used here with `cases`. I made the comments a bit less confusing.

If you meant something else, I'm happy to factor things out, but currently I can't see a part that would be useful as a separate lemma.